t/t1092-sparse-checkout-compatibility.sh

```diff
@@ -700,6 +700,60 @@ test_expect_success 'cherry-pick with conflicts' '
 	test_all_match test_must_fail git cherry-pick to-cherry-pick
 '
 
+test_expect_success 'checkout-index inside sparse definition' '
+	init_repos &&
+
+	run_on_all rm -f deep/a &&
+	test_all_match git checkout-index -- deep/a &&
```

Comment:
I've seen this `--` usage several times but am not quite sure what it does. Mind providing a quick explanation?

Reply:
The `--` signals the end of options and everything that remains should be parsed as plain-text input for a list of values. Typically, this means a list of pathspecs. It allows you to do something like `git checkout-index -- --looks-like-an-option` if you are silly enough to have `--looks-like-an-option` in your working tree.